src/shared/treeview/utils.ts

```diff
@@ -96,16 +96,33 @@ export function unboxTreeNode<T>(node: TreeNode, predicate: (resource: unknown)
  * would be passed in as-is.
  */
 export class TreeShim extends AWSTreeNodeBase {
```

Comment:
The shim was updated to support this new functionality. It's not a perfect recreation because `TestNode` can handle async rendering of the node itself while the shim would show `Loading...` on initialization.